packages/algolia-search/plugins/AlgoliaSearchResultsPlugin.tsx

```diff
@@ -0,0 +1,24 @@
+// eslint-disable-next-line import/no-extraneous-dependencies
+import { ProductItemsGridProps } from '@graphcommerce/magento-product'
+import { IfConfig, PluginProps } from '@graphcommerce/next-config'
+import dynamic from 'next/dynamic'
+import { useSearchRoute } from '../hooks/useSearchRoute'
+
+export const component = 'ProductListItemsBase'
+export const exported = '@graphcommerce/magento-product'
+export const ifConfig: IfConfig = 'demoMode'
+
+function AlgoliaSearchPlugin(props: PluginProps<ProductItemsGridProps>) {
+  const { Prev } = props
+  const search = useSearchRoute()
+
+  if (!search) return <Prev {...props} />
+
+  const AlgoliaResults = dynamic(
+    async () => (await import('../components/Filters/AlgoliaResults')).AlgoliaResults,
+  )
+
+  return <AlgoliaResults {...props} />
+}
+
+export const Plugin = AlgoliaSearchPlugin
```

Comment:
This means that we completely switch to client side rendering, am I right? Is there a way to also do SSR for this? Maybe execute the search in getStaticProps as well?

Reply:
The `react-instantsearch-hooks` library does support SSR, although it looks like we won't be able to implement this through plugins. 

https://www.algolia.com/doc/guides/building-search-ui/going-further/server-side-rendering

Reply:
We could add the SSR support and include some extra plugin installation steps to activate this feature. Adding this to the README!